tests/suites/test_suite_platform.function

```diff
@@ -0,0 +1,87 @@
+/* BEGIN_HEADER */
+
+/* This test module exercises the platform_* module. Since, depending on the
+ * underlying operating system, the time routines are not always reliable,
+ * this suite only performs very basic sanity checks of the timing API.
+ */
+
+#include <limits.h>
+
+#if defined(MBEDTLS_HAVE_TIME)
+#include "mbedtls/platform_time.h"
+
+#ifdef WIN32
+#include <windows.h>
+#elif _POSIX_C_SOURCE >= 199309L
+#include <time.h>
+#else
+#include <unistd.h>
+#endif
+void sleep_ms(int milliseconds)
+{
+#ifdef WIN32
+    Sleep(milliseconds);
+#elif _POSIX_C_SOURCE >= 199309L
+    struct timespec ts;
+    ts.tv_sec = milliseconds / 1000;
+    ts.tv_nsec = (milliseconds % 1000) * 1000000;
+    nanosleep(&ts, NULL);
+#else
+    usleep(milliseconds * 1000);
+#endif
+}
+#endif
+
+/* END_HEADER */
+
+/* BEGIN_DEPENDENCIES */
+
+/* END_DEPENDENCIES */
+
+
+
+/* BEGIN_CASE depends_on:MBEDTLS_HAVE_TIME */
+void time_get_milliseconds()
+{
+    mbedtls_ms_time_t  current = mbedtls_ms_time();
+    (void) current;
+    /* This goto is added to avoid warnings from the generated code. */
+    goto exit;
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_HAVE_TIME */
+void time_get_seconds()
+{
+    mbedtls_time_t  current = mbedtls_time(NULL);
+    (void) current;
+    /* This goto is added to avoid warnings from the generated code. */
+    goto exit;
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_HAVE_TIME */
+void time_delay_milliseconds(int delay)
+{
+    mbedtls_ms_time_t  current = mbedtls_ms_time();
+
+    sleep_ms(delay*1000);
+
+    current = (mbedtls_ms_time() - current) / 1000;
+    TEST_ASSERT(current == delay);
+    /* This goto is added to avoid warnings from the generated code. */
+    goto exit;
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_HAVE_TIME */
+void time_delay_seconds(int delay)
+{
+    mbedtls_time_t  current = mbedtls_time(NULL);
+    sleep_ms(delay*1000);
+    current = mbedtls_time(NULL) - current;
+    TEST_ASSERT(current == delay);
```

Comment:
Asserting that the elapsed time is exactly the delay expected isn't necessarily always going to be true, since on a CI there can be random additional delays. That's why I was very precise in my requirements: the elapsed time should be >= the delay, and (elapsed - delay) should be < some low value (here you are assuming < 1 will be sufficient; for "seconds" I might choose "< 4" and for "milliseconds" I might choose "< 500" - which is presumably why you had to change the "millsecond" test not to actually delay for a number of milliseconds

Reply:
Thanks. I am suffer from the fails.

I will revert to first version and add your suggestion.  above feedbacks might be invalid.

Reply:
And I just force push last commit. Sorry for that.